src/uct/ib/ud/base/ud_inl.h

```diff
@@ -98,16 +98,15 @@ static inline ucs_time_t uct_ud_fast_tick()
 }
 
 static UCS_F_ALWAYS_INLINE void
-uct_ud_am_set_zcopy_desc(uct_ud_send_skb_t *skb, const void *payload, size_t length, 
-                         uint32_t lkey, uct_completion_t *comp)
+uct_ud_am_set_zcopy_desc(uct_ud_send_skb_t *skb, const uct_iov_t *iov, size_t iovcnt,
+                         uct_completion_t *comp)
 {
     uct_ud_zcopy_desc_t *zdesc;
 
     skb->flags        |= UCT_UD_SEND_SKB_FLAG_ZCOPY;
     zdesc              = uct_ud_zcopy_desc(skb);
-    zdesc->lkey        = lkey;
-    zdesc->payload     = payload;
-    zdesc->len         = length;
+    zdesc->iov         = iov;
```

Comment:
this assumes IOV array itself has to remain valid until send completion, which is an incorrect assumption
need to make a copy of the IOV list inside "*zdesc". we can allocate zdesc size as the largest iov count we support.